generate.py

```diff
@@ -36,18 +44,30 @@ async def generate_conversations(
         Exception: Other errors
     """
     if verbose:
-        print("🔄 Generating conversations...")
+        print(f"🔄 Generating conversations with the following parameters:")
+        print(f"  - Persona model: {persona_model_config}")
+        print(f"  - Agent model: {agent_model_config}")
+        print(f"  - Persona extra run params: {persona_extra_run_params}")
+        print(f"  - Agent extra run params: {agent_extra_run_params}")
+        print(f"  - Max turns: {max_turns}")
+        print(f"  - Runs per prompt: {runs_per_prompt}")
+        print(f"  - Persona names: {persona_names}")
+        print(f"  - Folder name: {folder_name}")
+        print(f"  - Run ID: {run_id}")
     
     # Generate default folder name if not provided
-    if folder_name is None:
+    if run_id is None:
         timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
-        persona_meta = persona_model_config["model"].replace("-", "_").replace(".", "_")
-        agent_meta = agent_model_config["model"].replace("-", "_").replace(".", "_")
-        if extra_run_params:
-            run_id = f'p_{persona_meta}__a_{agent_meta}__t{max_turns}__r{runs_per_prompt}__{timestamp}__{extra_run_params}'
-        else:
-            run_id = f'p_{persona_meta}__a_{agent_meta}__t{max_turns}__r{runs_per_prompt}__{timestamp}'
-        folder_name = f"conversations/{run_id}"
+        persona_info = persona_model_config["model"].replace("-", "_").replace(".", "_")
+        agent_info = agent_model_config["model"].replace("-", "_").replace(".", "_")
+        if persona_extra_run_params:
+            persona_info += f"_{persona_extra_run_params}"
+            
+        if agent_extra_run_params:
+            agent_info += f"_{agent_extra_run_params}"
+        
+        run_id = f'p_{persona_info}__a_{agent_info}__t{max_turns}__r{runs_per_prompt}__{timestamp}'
```

Comment:
Should we replace "/"'s here with underscores? `os.makedirs` will handle the nested directories but not sure that's what you want. Could be relevant specially with `agent_extra_run_params` and `persona_extra_run_params` being added to this path.